docs/src/main/asciidoc/ansible.adoc

```diff
@@ -0,0 +1,238 @@
+
+First, because the collection integrated, we can check the status of the service on one of the targets:
+
+[source,bash]
+----
+# systemctl status getting-started.service
+● getting-started.service - A Quarkus service named getting-started
+   Loaded: loaded (/usr/lib/systemd/system/getting-started.service; enabled; vendor preset: disabled)
+   Active: active (running) since Thu 2023-04-13 12:48:18 UTC; 2min 40s ago
+ Main PID: 853 (java)
+	Tasks: 43 (limit: 1638)
+   Memory: 73.3M
+   CGroup: /system.slice/getting-started.service
+       	└─853 /usr/bin/java -jar /opt/quarkus_deploy/quarkus-run.jar
+
+Apr 13 12:48:18 bd71f39642c8 systemd[1]: Started A Quarkus service named getting-started.
+Apr 13 12:48:19 bd71f39642c8 java[853]: __  ____  __  _____   ___  __ ____  ______
+Apr 13 12:48:19 bd71f39642c8 java[853]:  --/ __ \/ / / / _ | / _ \/ //_/ / / / __/
+Apr 13 12:48:19 bd71f39642c8 java[853]:  -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \
+Apr 13 12:48:19 bd71f39642c8 java[853]: --\___\_\____/_/ |_/_/|_/_/|_|\____/___/
+Apr 13 12:48:19 bd71f39642c8 java[853]: 2023-04-13 12:48:19,284 INFO  [io.quarkus] (main) getting-started 1.0.0-SNAPSHOT on JVM (powered by Quarkus 2.16.6.Final) started in 0.607s. Listening on: http://0.0.0.0:8080
+Apr 13 12:48:19 bd71f39642c8 java[853]: 2023-04-13 12:48:19,309 INFO  [io.quarkus] (main) Profile prod activated.
+Apr 13 12:48:19 bd71f39642c8 java[853]: 2023-04-13 12:48:19,310 INFO  [io.quarkus] (main) Installed features: [cdi, resteasy-reactive, smallrye-context-propagation, vertx]
+----
+
+Manually, you can also test if the app is reachable:
+
+[source,bash]
+----
+# curl -I http://localhost:8080/
+HTTP/1.1 200 OK
+accept-ranges: bytes
+content-length: 3918
+cache-control: public, immutable, max-age=86400
+last-modified: Thu, 2 Mar 2023 11:03:18 GMT
+date: Thu, 2 Mar 2023 11:03:18 GMT
+----
+
+We'll see how to automate those validation in the next and last part of this tutorial.
+
+=== Writing a playbook
+
+Of course, you’ll most likely need to build on this, so you may want to write up your own playbook, rather than just using the one provided by the collection. Below is an example of such playbook:
+
+[source,yml]
+----
+- name: "Build and deploy a Quarkus app using Ansible"
+  hosts: all
+  gather_facts: false
+  vars:
+	quarkus_app_repo_url: 'https://github.com/quarkusio/quarkus-quickstarts.git'
+	app_name: getting-started
+	quarkus_app_source_folder: getting-started
+	quarkus_path_to_folder_to_deploy: /opt/quarkus_deploy
+
+  pre_tasks:
+	- name: "Build the Quarkus from {{ quarkus_app_repo_url }}."
+  	ansible.builtin.include_role:
+    	name: quarkus
+    	tasks_from: build.yml
+
+  tasks:
+	- name: "Deploy Quarkus app on target."
+  	ansible.builtin.include_role:
+    	name: quarkus
+    	tasks_from: deploy.yml
+----
+
+To run this playbook, you use again the ansible-playbook command, but providing now the path to the playbook:
+
+[source,bash]
+----
+$ ansible-playbook -i inventory playbook.yml
+----
+
```

Comment:
I'm trying to test with a playbook based on the previous template, but the execution failing.

The error is the following:

```
TASK [middleware_automation.quarkus.quarkus : Ensure Quarkus system group exists on target system] ***********************************************************************************
fatal: [127.0.0.1]: FAILED! => {"changed": false, "msg": "groupadd: Permission denied.\ngroupadd: cannot lock /etc/group; try again later.\n", "name": "quarkus_app"}
```

My playbook is this:

```
- name: "Build and deploy a Quarkus app using Ansible"
  hosts: all
  gather_facts: true
  collections:
    - middleware_automation.quarkus
  vars:
    quarkus_app_repo_url: https://github.redhat/quarkusio/quarkus-quickstarts.git
    app_name: getting-started
    quarkus_app_source_folder: getting-started
    quarkus_path_to_folder_to_deploy: /tmp/quarkus_deploy

  pre_tasks:
  - name: "Build the Quarkus from {{ quarkus_app_repo_url }}."
    ansible.builtin.include_role:
      name: quarkus
      tasks_from: build.yml

  tasks:
  - name: "Deploy Quarkus app on target."
    ansible.builtin.include_role:
      name: quarkus
      tasks_from: deploy.yml
```

I think the problem is at collection level and might be that the group creation is missing a `become: true`. - https://github.com/ansible-middleware/quarkus/blob/main/roles/quarkus/tasks/deploy.yml#L19

The same might happen with the `Ensure Quarkus user exists on target system.` task.

Reply:
Nice catch! Thanks! I sadly tested running Ansible as root, so I did not spot this issue.

Fixed in this [commit](https://github.com/ansible-middleware/quarkus/commit/8a7fee5b2a94bf63a3e60ab5c42ad4ef9a15b9f3)